src/main/java/games/strategy/triplea/delegate/AbstractPlaceDelegate.java

```diff
@@ -211,29 +211,11 @@ public String placeUnits(final Collection<Unit> units, final Territory at, final
     final List<Unit> unitsLeftToPlace = new ArrayList<>(units);
     unitsLeftToPlace.sort(getUnitConstructionComparator());
 
-    final boolean hasRemainingProduction = (unitsLeftToPlace.size() < maxPlaceableMap.totalValues());
     final List<Unit> remainingUnitsToPlace = new ArrayList<>(m_player.getUnits().getUnits());
     remainingUnitsToPlace.removeAll(unitsLeftToPlace);
-    final boolean hasRemainingNonConstructionUnitsToPlace =
-        Match.anyMatch(remainingUnitsToPlace, Matches.UnitIsNotConstruction);
     while (!unitsLeftToPlace.isEmpty() && !producers.isEmpty()) {
-
       // Get next producer territory
-      Territory producer = producers.get(0);
-      final boolean isCarrierLeftAndCanMoveExistingFightersToCarrier =
-          Match.anyMatch(unitsLeftToPlace, Matches.UnitIsCarrier) && canMoveExistingFightersToNewCarriers()
-              && !Properties.getLHTRCarrierProductionRules(getData());
-      final boolean cantAdjustUnitPlacements =
-          isUnitPlacementRestrictions() && Match.anyMatch(unitsLeftToPlace, Matches.UnitRequiresUnitsOnCreation);
-      if (producers.size() > 1
-          && (isCarrierLeftAndCanMoveExistingFightersToCarrier
-              || (hasRemainingProduction && hasRemainingNonConstructionUnitsToPlace && cantAdjustUnitPlacements))) {
-        producer = getRemotePlayer().selectProducerTerritoryForUnits(producers, at);
-        if (producer == null) {
-          break;
-        }
-      }
-      producers.remove(producer);
+      final Territory producer = producers.remove(0);
```

Comment:
I didn't do a straightforward revert to this file because there have been other changes made since #1766 that seem to rely on the new looping behavior added in #1766.  To keep this PR as simple as possible, I just selectively removed the lines that were affected after reverting the changes to `ITripleAPlayer`.